### PR TITLE
(SIMP-5085) Add unavailable options

### PR DESCRIPTION
**Pull request by jeefberkey**
- Add some previously impossible to configure options
  - su and su_user and su_group
  - the hourly rotate period
  - shred and shredcycles
- Added Logrotate::Periods type alias

SIMP-5085 #close

### CODE REVIEW

**Comment by jeefberkey** [comment]
https://gitlab.com/simp/pupmod-simp-logrotate/pipelines/31707872

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,10 @@
-* Mon Jul 16 2018 Trevor Vaughan <tvaughan@onyxpoint.com> - 6.2.1-0
+* Mon Oct 01 2018 Nick Miller <nick.miller@onyxpoint.com> - 6.2.1-0
```
Comment (lnemsick-simp) [inline CHANGELOG]: IMHO this version should be 6.3.0 because it adds features

File: README.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Manages the following:
 
 * `logrotate` package
 * `/etc/logrotate.conf` file
-* `/etc/logrotate.d` directory and its contents
+* `/etc/logrotate.simp.d` directory and its contents
```
Comment (lnemsick-simp) [inline README.md]: good catch

File: manifests/rule.pp [hunk 3]
```diff
@@ -64,48 +67,59 @@
 #   If undefined it defaults to the setting in logrotate
 # @param size
 # @param sharedscripts
+# @param shred
+# @param shredcycles
+# @param su
```
Comment (lnemsick-simp) [inline manifests/rule.pp]: a description of the su params would be nice as they do not one-for-one match a configuration item

File: spec/defines/rule_spec.rb [hunk 4]
```diff
@@ -1,77 +1,77 @@
...
-          it { is_expected.to create_file('/etc/logrotate.simp.d/test_logrotate_title').with_content(/test1\.log.*test2\.log/) }
-          it { is_expected.to_not create_file('/etc/logrotate.simp.d/test_logrotate_title').with_content(/lastaction/) }
-        end
+      context 'without a lastaction specified and lastaction_restart_logger = false' do
+        let(:params) {{
+          :log_files => ['test1.log', 'test2.log']
+        }}
+        let(:content) { File.read('spec/defines/expected/rule.txt') }
+        it { is_expected.to compile.with_all_deps }
+        # it { is_expected.to create_file('/etc/logrotate.simp.d/test_logrotate_title').with_content(/test1\.log.*test2\.log/) }
```
Comment (lnemsick-simp) [inline spec/defines/rule_spec.rb]: please remove commented out line

File: manifests/rule.pp [hunk 5]
```diff
@@ -64,48 +67,59 @@
...
+  Optional[Integer[0]]            $size                      = undef,
+  Boolean                         $sharedscripts             = true,
+  Optional[Boolean]               $shred                     = undef,
+  Optional[Integer[0]]            $shredcycles               = undef,
+  Boolean                         $su                        = false,
+  Optional[String[1]]             $su_user                   = undef,
+  Optional[String[1]]             $su_group                  = undef,
+  Integer[0]                      $start                     = 1,
+  Array[String[1]]                $tabooext                  = [],
 ) {
```
Comment (lnemsick-simp) [inline manifests/rule.pp]: Should have a check that if su is true, su_user and su_group are specified

File: spec/defines/rule_spec.rb [hunk 6]
```diff
@@ -1,77 +1,77 @@
...
-  context 'supported operating systems' do
-    on_supported_os.each do |os, facts|
-      context "on #{os}" do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
 
-        let(:title) {'test_logrotate_title'}
-        let(:pre_condition) { 'include "logrotate"' }
-        let(:facts) { facts }
+      let(:title) {'test_logrotate_title'}
```
Comment (lnemsick-simp) [inline spec/defines/rule_spec.rb]: where are the tests for the new params?

File: templates/conf.erb [hunk 7]
```diff
@@ -108,8 +111,20 @@
...
+<%    if @shredcycles -%>
+    shredcycles <%= @shredcycles.to_s %>
+<%    end -%>
+<%  end -%>
+<%  if @su && @su_user && @su_group -%>
+    su <%= @su_user %> <%= @su_group %>
+<%  end -%>
 }
-<% if @tabooext -%>
+<% unless (@tabooext.nil? || @tabooext.empty?) -%>
```
Comment (lnemsick-simp) [inline templates/conf.erb]: shouldn't the type for tabooext prevent it from being empty?

File: manifests/rule.pp [hunk 8]
```diff
@@ -64,48 +67,59 @@
 #   If undefined it defaults to the setting in logrotate
 # @param size
 # @param sharedscripts
+# @param shred
```
Comment (lnemsick-simp) [inline manifests/rule.pp]: why not add noshred, as well?

Comment (jeefberkey) [reply manifests/rule.pp]: Other params in the template that look like `feature` and `nofeature` set `feature` if the bool is true and set `nofeature` if the bool is false. I was hoping to make the behavior consistent.

File: templates/conf.erb (hunk 7 above)
Comment (jeefberkey) [reply templates/conf.erb]: Yes. will fix.

**Comment by jeefberkey** [comment]
@lnemsick-simp Should the define fail on el6 if su is specified or should it silently not set the option?

**Comment by jeefberkey** [comment]
https://gitlab.com/simp/pupmod-simp-logrotate/pipelines/32144719

**Comment by lnemsick-simp** [comment]
@jeefberkey I'm not sure if we have standard ways of handling this in other projects. I think you are going to have to ask the development community.

**Comment by jeefberkey** [comment]
@lnemsick-simp I think I prefer to ignore the option on EL6, and I've updated it to reflect that.

**Comment by jeefberkey** [comment]
https://gitlab.com/simp/pupmod-simp-logrotate/pipelines/32561071

**Comment by lnemsick-simp** [comment]
I'll add the missing tests

**Pull request merged**